phoenix-core/src/it/java/org/apache/phoenix/end2end/OnDuplicateKeyIT.java

```diff
@@ -720,6 +747,470 @@ public void testMultiplePartialUpdatesInSameBatch() throws Exception {
+
+            rs = conn.createStatement().executeQuery(dql);
+            assertTrue(rs.next());
+            assertEquals("abc", rs.getString("pk"));
+            assertEquals(1, rs.getInt("counter1"));
+            assertEquals(101, rs.getInt("counter2"));
+            assertFalse(rs.next());
+
+            List<Long> timestampList4 = getAllColumnsLatestCellTimestamp(conn, tableName);
+            assertTrue(timestampList4.get(0) > timestampList3.get(0)
+                    && timestampList4.get(1) > timestampList3.get(1)
+                    && timestampList4.get(2) > timestampList3.get(2));
+        }
+    }
+
+    @Test
+    @Ignore("Until Phoenix upgrades HBase dependency to 2.6.0 to include HBASE-28424")
+    public void testBatchedUpsertOnDupKeyAutoCommit() throws Exception {
+        testBatchedUpsertOnDupKey(true);
+    }
+
+    @Test
+    @Ignore("Until Phoenix upgrades HBase dependency to 2.6.0 to include HBASE-28424")
+    public void testBatchedUpsertOnDupKeyNoAutoCommit() throws Exception {
+        testBatchedUpsertOnDupKey(false);
+    }
+
+    private void testBatchedUpsertOnDupKey(boolean autocommit) throws Exception {
+        Properties props = PropertiesUtil.deepCopy(TEST_PROPERTIES);
+        String tableName = generateUniqueName();
+        try (Connection conn = DriverManager.getConnection(getUrl(), props)) {
+            conn.setAutoCommit(autocommit);
+
+            Statement stmt = conn.createStatement();
+
+            stmt.execute("create table " + tableName + "(pk varchar primary key, " +
+                    "counter1 integer, counter2 integer, approval varchar)");
+            createIndex(conn, tableName);
+
+            stmt.execute("UPSERT INTO " + tableName + " VALUES('a', 0, 10, 'NONE')");
+            conn.commit();
+
+            stmt.addBatch("UPSERT INTO " + tableName +
+                    " (pk, counter1, counter2) VALUES ('a', 0, 10) ON DUPLICATE KEY IGNORE");
+            stmt.addBatch("UPSERT INTO " + tableName +
+                    " (pk, counter1, counter2) VALUES ('a', 0, 10) ON DUPLICATE KEY UPDATE" +
+                    " counter1 = CASE WHEN counter1 < 1 THEN 1 ELSE counter1 END");
+
+            stmt.addBatch("UPSERT INTO " + tableName +
+                    " (pk, counter1, counter2) VALUES ('b', 0, 9) ON DUPLICATE KEY IGNORE");
+            String dml =  "UPSERT INTO " + tableName +
+                    " (pk, counter1, counter2) VALUES ('b', 0, 10) ON DUPLICATE KEY UPDATE" +
+                    " counter2 = CASE WHEN counter2 < 11 THEN counter2 + 1 ELSE counter2 END," +
+                    " approval = CASE WHEN counter2 < 10 THEN 'NONE'" +
+                    " WHEN counter2 < 11 THEN 'MANAGER_APPROVAL'" +
+                    " ELSE approval END";
+            stmt.addBatch(dml);
+            stmt.addBatch(dml);
+            stmt.addBatch(dml);
+
+            int[] actualReturnValues = stmt.executeBatch();
+            int[] expectedReturnValues = new int[]{0, 1, 1, 1, 1, 0};
```

Comment:
I don't think this is correct for the case when auto-commit is true and batch API is used. AFAIK for batching with/without auto-commit the return values will always be {1,1,....1}

Reply:
@tkhurana With auto-commit set to true, the size of mutationBatch will be 1 each time,  so the `returnCodes` will be updated by `statement.getUpdateCount()` for each statement in `PhoenixStatement#executeBatch()`.

Reply:
@jinggou PhoenixStatement#executeBatch internally turns off auto-commit

Reply:
@jinggou So when will it return 0 ?

Reply:
Won't getUpdateCount() always return 1 ? We are not changing that API right ?

Reply:
@tkhurana We didn't change `PhoenixStatement#getUpdateCount()` API, but since it calls `PhoenixStatement#getLastUpdateCount()`, 
```
    @Override
    public int getUpdateCount() throws SQLException {
        int updateCount = getLastUpdateCount();
        // Only first call can get the update count, otherwise
        // some SQL clients get into an infinite loop when an
        // update occurs.
        setLastUpdateCount(NO_UPDATE);
        return updateCount;
    }
```
while `lastUpdateCount` will be changed and `setLastUpdateCount` in our new logic:
```
                                int lastUpdateCount = (int) Math.min(Integer.MAX_VALUE,
                                        lastState.getUpdateCount());
                                if (connection.getAutoCommit()) {
                                    connection.commit();
                                    if (isAtomicUpsert) {
                                        lastUpdateCount = connection.getMutationState()
                                                .getNumUpdatedRowsForAutoCommit();
                                    }
                                }
                                ...
                                setLastUpdateCount(lastUpdateCount);
```
I think that's why it won't always return 1.
FYI @virajjasani

Reply:
@tkhurana Now I see that PhoenixStatement#executeBatch internally turns off auto-commit, and the code of `PhoenixStatement#executeBatch` is different in our lightfork. And I'm testing using lightfork, that's why this test will pass.

Reply:
I'll sync to the current `PhoenixStatement#executeBatch` and change the expected value to all 1's and test again.